packages/org.openapis.v3.contrib/SchemaGenerator.pkl

```diff
@@ -275,25 +284,50 @@ local function convertTypeForProperty(typ: reflect.Type, seenClasses: Set<Class>
 local function convertDeclarationForProperty(declaration: reflect.Declaration, seenClasses: Set<Class>): Schema.PropertySchema =
   convertDeclaration(declaration, seenClasses).toDynamic().toTyped(Schema.PropertySchema)
 
+local function isSameType(typeA: reflect.Type, typeB: reflect.Type) =
+  if (typeA is reflect.DeclaredType && typeB is reflect.DeclaredType)
+    typeA.referent.reflectee == typeB.referent.reflectee
+  else if (typeA is reflect.NullableType && typeB is reflect.NullableType)
+    isSameType(typeA.member, typeB.member)
+  else if (typeA is reflect.NothingType && typeB is reflect.NothingType)
+    true
+  else if (typeA is reflect.UnknownType && typeB is reflect.UnknownType)
+    true
+  else if (typeA is reflect.StringLiteralType && typeB is reflect.StringLiteralType)
+    typeA.value == typeB.value
+  // union types turn into Go's `any`, so we can say that this is always fine.
+  else if (typeA is reflect.UnionType && typeB is reflect.UnionType)
+    true
+  // remaining types: `FunctionType`, `TypeParameter`, `ModuleType`.
+  // we can actually check if `ModuleType` refers to the same type by checking if the enclosing declaration is the same,
+  // but we will pretend it is always false for now.
+  else false
+
 /// Given a class, return all of its properties, including properties of all inherited classes.
 ///
-/// Excludes properties that are functions, and any hidden properties. Also, excludes properties on [Module].
+/// Excludes properties that are functions, and any hidden properties.
 ///
 /// **NOTE**: JSON Schema has a way to combine schemas together using `allOf`. At first glance, this might
 /// seem like a good way to model inheritance; the schema is all of the child properties and the parent
 /// properties. However, this mechanism breaks if a child class overrides the property of a parent class.
 /// If an overridden property is mutually exclusive with the parent property, `allOf` would produce an
 /// invalid JSON schema because it is impossible for any object to match all of the specified constraints.
-local function getProperties(clazz: reflect.Class): Map<String, reflect.Property> =
-  // Edge case: Modules implicitly inherit the `Module` class. We do not want to capture this as part of the
-  // schema.
-  if (clazz.reflectee == Module)
-    Map()
+local function getProperties(clazz: reflect.Class) = getAllProperties(clazz)
```

Comment:
nit: why remove the return type annotation?

Reply:
Ah, no reason, this was just lost in the refactor.